Firmware/RoboPad/RoboPad.ino

```diff
@@ -423,22 +432,64 @@ void setup() {
   // Set up the websocket server:
   webSocketServer.begin();
   webSocketServer.onEvent(webSocketEvent);
+  Serial.println("WebSocket server started.");
 
   // Configure IO
   configureIO();
+
+  attachInterrupt(PPM_PIN, interruptPPM, CHANGE);
+  Serial.println("PPM interrupt attatched");
 }
 
+#ifdef PPM
+long counter = 0;
+long startMicros = 0;
+long minSync = 2500;
+bool ppmActive = false;
+
+ICACHE_RAM_ATTR void interruptPPM () {
+  if (digitalRead(PPM_PIN)) {
+    startMicros = micros();
+  } else {
+    long duration = micros() - startMicros;
+    if (duration > minSync) {
+      counter = 0;
+    } else {
+      channels[counter] = map(duration, PPM_MIN_PULSE_DURATION, PPM_MAX_PULSE_DURATION, 0.0, 255.0);
+      counter += 1;
+    }
+  }
+
+  ppmActive = true;
+}
+#endif
+
 BasicHBridgeMotor leftMotor(14,15);
 BasicHBridgeMotor rightMotor(12,13);
 void loop() {
   httpServer.handleClient();
 
   if(safeboot) return;// Make sure we don't do anything when safeboot is operational.
+
+  #ifdef PPM
+  if (ppmActive) {
+    esc.writeMicroseconds(map(channels[2], 0, 255, 1000, 2000));
+    leftMotor.drive(channels[0], 0, 255);
+    rightMotor.drive(channels[1], 0, 255);
```

Comment:
Haven't had to deal with it in a while, but if `channels` is made volatile then I _think_ this should be wrapped in an ATOMIC_BLOCK.

Reply:
I'm not familiar with ATOMIC_BLOCK. If you want to add it feel free but it seems to work fine without it.

Reply:
No probs, I've merged now - thanks for contributing! I'll add an issue to look into ATOMIC_BLOCK use, but if it works then it works :)